pallets/liquidity-provider/src/lib.rs

```diff
@@ -47,7 +45,11 @@ decl_module! {
         fn deposit_event() = default;
 
         #[weight = 0]
-        pub fn attest(origin, asset_id: CurrencyIdOf<T>, balance: BalanceOf<T>)   -> dispatch::DispatchResult{
+        pub fn attest(origin, asset_id: ValiuCurrencies, balance: BalanceOf<T>) -> dispatch::DispatchResult {
+            if let ValiuCurrencies::Usdv = asset_id {
+                return Err(Error::<T>::MustNotBeUsdv.into());
+            }
```

Comment:
This is a bit of a hack. I still need to find a better way to handle this properly

Reply:
I think it's fine for now :) Later(next ticket?) we could group currencies in different ways based on the usage and there wouldn't be a need for a check if the type is explicit about the allowed currency. I'm not sure if the SCALE encoder supports it but our currencies enum can be an union of other enums, so when we do need to work with all the currencies we have something that groups them but when we don't we use the specific variant we need.
```rust
enum Asset {
    USDv,
    OffChain(OffChainAsset), // for attestations
    Collateral(Collateral) // assets valid for use as USDv collateral?
    // or Fiat(Fiat), Digita(Digital)
    // ... or other ways of grouping? 🤷🏽‍♂️ 
}
```

Reply:
Nice idea!